pkg/activator/certificate/cache.go

```diff
@@ -101,9 +105,34 @@ func (cr *CertCache) updateCache(secret *corev1.Secret) {
 	}
 	pool.AddCert(ca)
 
-	cr.TLSConf.RootCAs = pool
-	cr.TLSConf.ServerName = certificates.LegacyFakeDnsName
-	cr.TLSConf.MinVersion = tls.VersionTLS12
+	cr.ClientTLSConf.RootCAs = pool
+	cr.ClientTLSConf.ServerName = certificates.LegacyFakeDnsName
+	cr.ClientTLSConf.MinVersion = tls.VersionTLS12
+	cr.ClientTLSConf.Certificates = []tls.Certificate{cert}
+
+	cr.ServerTLSConf.MinVersion = tls.VersionTLS12
+	cr.ServerTLSConf.GetCertificate = cr.GetCertificate
+
+	switch cr.trustConfig {
+	case netcfg.TrustIdentity, netcfg.TrustMutual:
+		cr.ServerTLSConf.ClientAuth = tls.RequireAndVerifyClientCert
+		cr.ServerTLSConf.ClientCAs = pool
+		cr.ServerTLSConf.VerifyConnection = func(cs tls.ConnectionState) error {
+			for _, match := range cs.PeerCertificates[0].DNSNames {
+				// Activator currently supports a single routingId which is the default "0"
+				// Working with other routingId is not yet implemented
+				if match == certificates.DataPlaneRoutingName("0") {
+					return nil
+				}
+				//Until all ingresses work with updated dataplane certificates  - allow also any legacy certificate
+				if match == certificates.LegacyFakeDnsName {
+					return nil
+				}
```

Comment:
Doesn't the default verification assert this already?

Reply:
The default verification checks for LegacyFakeDnsName.

This code checks for DataPlaneRoutingName("0"). 
Once ingresses all move to use the new certificates that include DataPlaneRoutingName("0"), we will remove the extra:
```
if match == certificates.LegacyFakeDnsName {
	return nil
}
```

Reply:
We should just drop `certificates.LegacyFakeDnsName` and not worry about doing this seamlessly.

Reply:
We can't do that, we rely on it in different places. 
We may be able to do that sometime in the future.

Reply:
Ok, it seems we are discussing here different aspects. 
See my comment below - we continue to support "internal encryption" via "trust=minimal"

As for line 128, I will remove it in a separate PR once we moved all ingresses to use the new certificates and we start cleaning up and removing internal-encryption as a config option. I would prefer to do that when we have e2e tests also. We are not yet there.